### PR TITLE
Do not delete old logs on try to add a log larger than the maximum storage capacity

### PR DESCRIPTION
**Pull request by MatkovIvan**
* [x] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Do not delete old logs on try to add a log larger than the maximum storage capacity.

## Related PRs or issues

https://github.com/Microsoft/AppCenter-SDK-Apple/pull/1204

### CODE REVIEW

**Comment by jaelim-ms** [comment]
@MatkovIvan Any related work items for this change?

**Comment by MatkovIvan** [comment]
Discussing offline, closing for now

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 ## Version 1.10.0 (Under active development)
 
+### AppCenter
```
Comment (guperrot) [inline CHANGELOG.md]: Will need to wait release to move the notes

**Comment by guperrot** [comment]
After merge back to develop after release, we'll restore the changelog message to this PR. That's why we don't merge it now.

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/persistence/DatabasePersistence.java [hunk 2]
```diff
@@ -271,6 +272,11 @@ public long putLog(@NonNull Log log, @NonNull String group, @IntRange(from = Fla
                 targetKey = null;
                 targetToken = null;
             }
+            long maxSize = mDatabaseManager.getMaxSize();
+            if (!isLargePayload && maxSize < payloadSize) {
```
Comment (jaelim-ms) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/persistence/DatabasePersistence.java]: We definitely know maxSize == payloadSize will drop all logs due to other information stored with the log. `<=` in the condition?

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/storage/DatabaseManager.java [hunk 3]
```diff
@@ -221,22 +220,25 @@ public long put(@NonNull ContentValues values, @NonNull String priorityColumn) {
                 } catch (SQLiteFullException e) {
 
                     /* Delete the oldest log. */
+                    AppCenterLog.debug(LOG_TAG, "Storage full, trying to delete oldest log with lowest priority...");
```
Comment (jaelim-ms) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/storage/DatabaseManager.java]: `Storage full` -> `Storage is full`
`lowest` -> `equal or lower`

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/persistence/DatabasePersistence.java (hunk 2 above)
Comment (jaelim-ms) [reply sdk/appcenter/src/main/java/com/microsoft/appcenter/persistence/DatabasePersistence.java]: The change is made [here](https://github.com/Microsoft/AppCenter-SDK-Apple/pull/1222) for Apple SDK.

**Pull request merged**